homeassistant/components/tplink/strings.json

```diff
@@ -136,5 +140,16 @@
         }
       }
     }
+  },
+  "options": {
+    "step": {
+      "init": {
+        "data": {
+          "tplink_cloud_username": "Enter your TPLink Cloud username",
+          "tplink_cloud_password": "Enter your TPLink Cloud password",
+          "tplink_cloud_credentials_sync": "Keep these credentials synced across devices"
```

Comment:
As this is a config entry specific setting, I have my doubts if it would be acceptable for homeassistant inclusion. How about doing it like this instead of syncing:
1. If the device works as-is, nothing to do
2. If the auth exception gets raised, ask user to input the creds
3. On success, store inside `self.hass[DOMAIN]` _and_ trigger reauth on other devices with the same error

This way, the creds are 1) stored in the config entry per-device basis, 2) there is no need for special syncing option, and 3) the user needs to supply the correct creds only once per session.

What do you think?